### DISCUSSION TITLE
Understanding Kernel

### DISCUSSION THREAD

**Discussion by nilo85** [Joshua-Riek/ubuntu-rockchip#729] [Q&A] [answered]
Hello, I have not done kernel hacking for some years (2.4 was the stable last time I touched it and was a LFS / Gentoo nerd)

I suffer from https://github.com/Joshua-Riek/ubuntu-rockchip/issues/606 that stops me from using the Opi5+ for my project, and as running foreign binaries is a no go for me, even if I got it running with "vendor ubuntu", I can't use that, so I am starting to investigate what is the state of the kernel.

Main kernel for use here is found here: https://github.com/Joshua-Riek/linux-rockchip 
then we have a kernel called mainline here https://github.com/Joshua-Riek/linux

I see the mainline has quite some commits ahead of Linus mirror, so I suspect it is not a mainline at all? is it mainline + patching in rockchip stuff?

What about the  linux-rockchip ? I realized the kernel provided by orangepi is some sort of android monster.

I guess what I am asking for is general guidance how to navigate the current jungle, so that I can target any efforts I put on troubleshooting the issue I have, so if anyone could help me sort out the state of the kernels and or context / background I think that could help anyone who wants to involve in this project.

Kind regards
NIklas

**Comment by stevepringle** [comment, marked as answer]
﻿The current state of mainline kernel support for the rk3588 can be seen here:-mainline-status.md · main · hardware-enablement / Rockchip RK3588 upstream enablement efforts / Notes for Rockchip 3588 · GitLabgitlab.collabora.comIf you want to experiment with the mainline kernel, the collabora version with all the support to date is here:-hardware-enablement / Rockchip RK3588 upstream enablement efforts / linux · GitLabgitlab.collabora.comRegards,SteveSent from my iPadOn 20 Apr 2024, at 09:57, Niklas Lönn ***@***.***> wrote:﻿
Hello, I have not done kernel hacking for some years (2.4 was the stable last time I touched it and was a LFS / Gentoo nerd)
I suffer from #606 that stops me from using the Opi5+ for my project, and as running foreign binaries is a no go for me, even if I got it running with "vendor ubuntu", I can't use that, so I am starting to investigate what is the state of the kernel.
Main kernel for use here is found here: https://github.com/Joshua-Riek/linux-rockchip
then we have a kernel called mainline here https://github.com/Joshua-Riek/linux
I see the mainline has quite some commits ahead of Linus mirror, so I suspect it is not a mainline at all? is it mainline + patching in rockchip stuff?
What about the  linux-rockchip ? I realized the kernel provided by orangepi is some sort of android monster.
I guess what I am asking for is general guidance how to navigate the current jungle, so that I can target any efforts I put on troubleshooting the issue I have, so if anyone could help me sort out the state of the kernels and or context / background I think that could help anyone who wants to involve in this project.
Kind regards
NIklas

**Reply by nilo85** [reply]
@stevepringle ow, awesome little project with nice todolists, sounds like your mainline is close to support what I need so I will try it out and see if I am capable an/or has time to do some kernel hacking coming days.
